plugins/modules/dcnm_image_upgrade.py

```diff
@@ -0,0 +1,5313 @@
+        - success:
+            - False if RETURN_CODE != 200 or MESSAGE != "OK"
+            - True otherwise
+        """
+        result = {}
+        if response.get("MESSAGE") != "OK":
+            result["success"] = False
+            result["changed"] = False
+            return result
+        if response.get("ERROR"):
+            result["success"] = False
+            result["changed"] = False
+            return result
+        result["success"] = True
+        result["changed"] = True
+        return result
+
+    def log_msg(self, msg):
+        """
+        used for debugging. disable this when committing to main
+        by setting __init__().debug to False
+        """
+        if self.debug is False:
+            return
+        if self.fd is None:
+            try:
+                self.fd = open(f"{self.logfile}", "a+", encoding="UTF-8")
+            except IOError as err:
+                msg = f"error opening logfile {self.logfile}. "
+                msg += f"detail: {err}"
+                self.module.fail_json(msg)
+
+        self.fd.write(msg)
+        self.fd.write("\n")
+        self.fd.flush()
+
+    def make_boolean(self, value):
+        """
+        Return value converted to boolean, if possible.
+        Return value, if value cannot be converted.
+        """
+        if isinstance(value, bool):
+            return value
+        if isinstance(value, str):
+            if value.lower() in ["true", "yes"]:
+                return True
+            if value.lower() in ["false", "no"]:
+                return False
+        return value
+
+    def make_none(self, value):
+        """
+        Return None if value is an empty string, or a string
+        representation of a None type
+        Return value otherwise
+        """
+        if value in ["", "none", "None", "NONE", "null", "Null", "NULL"]:
+            return None
+        return value
+
+
+class NdfcAnsibleImageUpgrade(NdfcAnsibleImageUpgradeCommon):
+    """
+    Ansible support for image policy attach, detach, and query.
+    """
+
+    def __init__(self, module):
+        super().__init__(module)
+        self.class_name = self.__class__.__name__
+        self.log_msg(f"{self.class_name}.__init__")
+        # populated in self._build_policy_attach_payload()
+        self.payloads = []
+
+        self.config = module.params.get("config")
+        self.log_msg(f"{self.class_name}.__init__() self.config {self.config}")
+        if not isinstance(self.config, dict):
+            msg = "expected dict type for self.config. "
+            msg = +f"got {type(self.config).__name__}"
+            self.module.fail_json(msg)
+
+        self.check_mode = False
+        self.validated = []
+        self.have_create = []
+        self.want_create = []
+        self.need = []
+        self.diff_save = {}
+        self.query = []
+        self.result = dict(changed=False, diff=[], response=[])
+
+        self.mandatory_global_keys = {"switches"}
```

Comment:
Thanks Mallik.  This is because policy name can be either in the global config or in individual switch configs.  So, it's mandatory, but can be excluded in the global config if the user has added it to the switch configs.